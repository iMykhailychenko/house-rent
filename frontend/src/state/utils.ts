import { ErrorState } from './interfaces/common';

export const formatSeverError = (error: Error & { status?: number; message?: string }): ErrorState => ({
    status: error.status,
    message: error.message || 'something_went_wrong',
});
