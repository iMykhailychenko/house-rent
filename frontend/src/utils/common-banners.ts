import { Banner, BannerType } from '../state/entities/banners/banners.interface';

export const POST_IN_ARCHIVE: Banner = {
    id: 'POST_IN_ARCHIVE',
    type: BannerType.ERROR,
    content:
        'УВАГА! Цей пост знаходиться в архіві. Щоб інші користувачі могли його побачити вам потрібно змінити статус цього посту на активний!',
};

export const POST_IN_DRAFT: Banner = {
    id: 'POST_IN_DRAFT',
    type: BannerType.INFO,
    content:
        'УВАГА! Цей пост не активований. Щоб інші користувачі могли його побачити вам потрібно змінити статус цього посту на активний!',
};

export const VALIDATE_EMAIL_WARN: Banner = {
    id: 'VALIDATE_EMAIL_WARN',
    type: BannerType.WARNING,
    content:
        'Ви не активували вашу електронну пошту. Щоб повною мірою використовувати функціонал сайту перейдіть за посиланням в листі',
};

export const VALIDATE_EMAIL_ERROR: Banner = {
    id: 'VALIDATE_EMAIL_ERROR',
    type: BannerType.ERROR,
    content:
        'Ви не активували вашу електронну пошту. Щоб повною мірою використовувати функціонал сайту перейдіть за посиланням в листі',
};
