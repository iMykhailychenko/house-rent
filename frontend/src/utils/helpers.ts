import cookie from 'cookie';

interface ParseCookieParams<T> {
    value?: string;
    key: string;
    isJson?: boolean;
    defaultValue: T;
}

export const parseCookie = <T>({ value = '', key, isJson = false, defaultValue }: ParseCookieParams<T>): T => {
    try {
        return isJson ? cookie.parse(value)[key] || defaultValue : JSON.parse(cookie.parse(value)[key]);
    } catch (error) {
        return defaultValue;
    }
};
