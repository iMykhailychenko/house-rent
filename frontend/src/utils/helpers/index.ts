export { copyText } from './copy-text.helper';
export { addMonthToDate, formatDate, formatTime, validateDate, onlineStatus } from './date.helper';
export { cutString } from './string.helper';
export { parseCookie } from './cookie.helper';
export { addZero } from './number.helper';
export { shallowRouter } from './shallow-router.helper';
export { errorNotif } from './error-logger.helper';
