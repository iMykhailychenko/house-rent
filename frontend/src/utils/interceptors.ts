import axios from 'axios';
import * as AxiosLogger from 'axios-logger';

const api = axios.create({});

api.interceptors.request.use(
    req => AxiosLogger.requestLogger(req, { data: false }),
    error => Promise.reject(error),
);

api.interceptors.response.use(
    res => AxiosLogger.responseLogger(res, { data: false }),
    error => {
        if (error?.response?.status === 401) {
            // delete api.defaults.headers.common.Authorization;
        }

        return Promise.reject(error);
    },
);

export default api;
