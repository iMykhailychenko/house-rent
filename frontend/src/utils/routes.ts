const routes = {
    home: '/',
    private: '/private',
    chat: '/chat',
    auth: {
        login: '/auth/login',
        join: '/auth/join',
        success: '/auth/success',
        reset: '/auth/reset',
    },
    users: {
        profile: (id: string | number): string => `/users/${id}`,
    },
    posts: {
        list: '/posts',
        new: '/posts/new',
        single: (id: string | number): string => `/posts/${id}`,
    },
};

export default routes;
