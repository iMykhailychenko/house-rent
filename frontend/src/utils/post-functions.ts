import { NextRouter } from 'next/router';
import { toast } from 'react-toastify';

import { modal } from '../components/common/modal/modal';
import toastConfig from '../config/toast.cofig';
import { ThunkAppDispatch } from '../hooks/redux.hook';
import { POST_STATUS } from '../state/entities/posts/posts.interface';
import { personalPostsListThunk } from '../state/entities/posts/thunks/personal-posts.thunk';
import { updatePostStatusThunk } from '../state/entities/posts/thunks/update-status.thunk';

import routes from './routes';

export enum POST_ACTIONS {
    ARCHIVE = 'archive',
    EDIT = 'edit',
    ACTIVATE = 'activate',
    DELETE = 'delete',
}

type PostActionsMap = { [key in POST_STATUS]: POST_ACTIONS[] };
export const postActionsMap: PostActionsMap = {
    [POST_STATUS.DRAFT]: [POST_ACTIONS.ACTIVATE, POST_ACTIONS.EDIT, POST_ACTIONS.ARCHIVE, POST_ACTIONS.DELETE],
    [POST_STATUS.ACTIVE]: [POST_ACTIONS.ARCHIVE, POST_ACTIONS.EDIT, POST_ACTIONS.DELETE],
    [POST_STATUS.ARCHIVE]: [POST_ACTIONS.ACTIVATE, POST_ACTIONS.DELETE, POST_ACTIONS.EDIT],
};

type PostFunctionsMap = { [key in POST_ACTIONS]: (postId: number) => Promise<void> };
export const postFunctionsMap = (dispatch: ThunkAppDispatch, history: NextRouter): PostFunctionsMap => {
    const page = +String(history.query.page || 1);
    const status = String(history.query.status || POST_STATUS.DRAFT) as POST_STATUS;

    return {
        [POST_ACTIONS.ARCHIVE]: async (postId: number): Promise<void> => {
            modal.close();
            const post = await dispatch(updatePostStatusThunk({ id: postId, status: POST_STATUS.ARCHIVE })).unwrap();
            if (!post.id) return;
            await dispatch(personalPostsListThunk({ status, page }));
            toast.success('Ви успішно перемістили пост до Архіву. Тепер він прихований від інших користувачів', toastConfig);
        },
        [POST_ACTIONS.EDIT]: async (postId: number): Promise<void> => {
            modal.close();
            history.push(routes.posts.edit(postId));
        },
        [POST_ACTIONS.ACTIVATE]: async (postId: number): Promise<void> => {
            modal.close();
            const post = await dispatch(updatePostStatusThunk({ id: postId, status: POST_STATUS.ACTIVE })).unwrap();
            if (!post.id) return;
            await dispatch(personalPostsListThunk({ status, page }));
            toast.success('Ви успішно активували свій пост! Тепер він доступний для інших користувачів на сайті', toastConfig);
        },
        [POST_ACTIONS.DELETE]: async (postId: number): Promise<void> => {
            modal.close();
            const post = await dispatch(updatePostStatusThunk({ id: postId, status: POST_STATUS.ACTIVE })).unwrap();
            if (!post.id) return;
            await dispatch(personalPostsListThunk({ status, page }));
            toast.success('Ви успішно активували свій пост! Тепер він доступний для інших користувачів на сайті', toastConfig);
        },
    };
};
