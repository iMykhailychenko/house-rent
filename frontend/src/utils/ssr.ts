import { Store } from '@reduxjs/toolkit';
import axios from 'axios';
import { GetServerSideProps, GetServerSidePropsContext, Redirect } from 'next';

import { cookiesTokenAction } from '../state/entities/auth/auth.reducer';
import { profileInfoThunk } from '../state/entities/profile/profile.thunk';
import rootInitialState from '../state/initial-state';
import { RootState } from '../state/reducer';
import { initializeStore } from '../state/store';

import routes from './routes';

type ReturnCallback<T> = { props: T | { state: RootState } };
type SSRCallback<T> = (
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    context: GetServerSidePropsContext & { store?: Store<RootState, any> },
) => Promise<ReturnCallback<T> | void>;

export const withStore = <T>(callback?: SSRCallback<T>): GetServerSideProps => {
    return async (context: GetServerSidePropsContext): Promise<ReturnCallback<T>> => {
        const store = initializeStore(rootInitialState);

        const token = axios.defaults.headers.common.Authorization;
        if (token) {
            await store.dispatch(profileInfoThunk());
            await store.dispatch(cookiesTokenAction(token));
        }

        if (callback) {
            const result = await callback({ ...context, store });
            return { props: { ...(result?.props || {}), state: store.getState() } };
        }

        return { props: { state: store.getState() } };
    };
};

export const withAuthRedirect =
    <T>(callback: SSRCallback<T> | null, reverse = false): GetServerSideProps =>
    async (context: GetServerSidePropsContext): Promise<ReturnCallback<T> | { redirect: Redirect }> => {
        const token = axios.defaults.headers.common.Authorization;
        if (reverse ? token : !token) {
            return {
                redirect: {
                    statusCode: 302,
                    destination: routes.auth.login,
                },
            };
        }

        const store = initializeStore(rootInitialState);

        if (callback) {
            const result = await callback({ ...context, store });
            return { props: { ...(result?.props || {}), state: store.getState() } };
        }

        return { props: { state: store.getState() } };
    };
