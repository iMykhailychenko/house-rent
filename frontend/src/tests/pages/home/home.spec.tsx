import Home from '../../../pages';
import { render } from '../../tests.utils';

describe('Home page', () => {
    it('matches snapshot', () => {
        const { asFragment } = render(<Home />);
        expect(asFragment()).toMatchSnapshot();
    });
});
