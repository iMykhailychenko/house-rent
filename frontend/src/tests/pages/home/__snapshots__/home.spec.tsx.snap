// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <div
      class="inner md root"
    >
      <div
        class="item"
      >
        <a
          class="home button secondary"
          href="/"
        >
          <span
            class="children"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              />
            </svg>
          </span>
        </a>
        <button
          class="button button menu"
          type="button"
        >
          <span
            class="children"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
        </button>
        <div
          class="root"
        >
          <button
            class="button active"
            type="button"
          >
            ua
          </button>
          <button
            class="button"
            type="button"
          >
            ru
          </button>
        </div>
      </div>
      <div
        class="item"
      >
        <button
          class="button root"
          type="button"
        >
          <span
            class="children"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6zm0-10c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4z"
              />
            </svg>
          </span>
        </button>
        <div
          class="root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
          <button
            class="button"
            type="button"
          >
            увійти
          </button>
          <button
            class="button"
            type="button"
          >
            зареєструватися
          </button>
        </div>
      </div>
    </div>
  </header>
  <h1
    class="title"
  >
    Ми перевернули ринок нерухомості з ніг на голову. Тепер орендарі публікують оголошення, а рієлтори та власники пропонують їм житло.
  </h1>
  <section
    class="root section"
  >
    <div
      class="lg root"
    >
      <div
        class="root tenant"
      >
        <div
          class="inner"
        >
          <button
            class="root switch"
            type="button"
          >
            <span
              class="active"
              style="min-width: 21rem;"
            >
              Орендарям
            </span>
            <span
              class=""
              style="min-width: 21rem;"
            >
              Власникам / рієлторам
            </span>
          </button>
          <div
            class="animation"
          >
            <h2
              class="title-1 appear appear-active"
            >
              Досить витрачати свій час на пошук житла! Твоя квартира знайде тебе сама.
            </h2>
            <p
              class="text-1 appear appear-active"
            >
              Набридло годинаму шукати квартиру, знаходити підходящій варінт, а в результаті дізнаватися що вона вже здана? Ми перевернули все з ніг на голову. Тепер ви публікуєте оголошення, а рієлтори та власники пропонують вам житло
            </p>
            <a
              class="button primary appear appear-active"
              href="/posts/new"
            >
              <span
                class="children"
              >
                Створити оголошення
              </span>
            </a>
          </div>
        </div>
        <img
          alt=""
          class="img appear appear-active"
          src="/pages/home/tenant.png"
        />
      </div>
    </div>
  </section>
  <section
    class="root"
    id="home-posts"
  >
    <div
      class="md root"
    >
      <div
        class="root"
      >
        <h2
          class="title-2"
        >
          Якщо ви рієлтори чи власник квартири, то в першу чергу зверніть увагу на термінові оголошення
        </h2>
        <div
          class="flex"
        >
          <div
            class="lg"
          />
          <aside
            class="root"
          >
            <div
              class="sticky"
            >
              <div
                class="scrollBox"
              >
                <div
                  class="flex"
                >
                  <button
                    class="button active"
                    title="large cards"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M4 15h16v-2H4v2zm0 4h16v-2H4v2zm0-8h16V9H4v2zm0-6v2h16V5H4z"
                      />
                    </svg>
                  </button>
                  <span
                    class="separator"
                  />
                  <button
                    class="button"
                    title="small cards"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M4 8h4V4H4v4zm6 12h4v-4h-4v4zm-6 0h4v-4H4v4zm0-6h4v-4H4v4zm6 0h4v-4h-4v4zm6-10v4h4V4h-4zm-6 4h4V4h-4v4zm6 6h4v-4h-4v4zm0 6h4v-4h-4v4z"
                      />
                    </svg>
                  </button>
                </div>
                <div
                  class="root"
                >
                  <div
                    class="flex"
                  >
                    <button
                      class="chip"
                      style="--hover: #cf1322;"
                      title="hot"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M13.5.67s.74 2.65.74 4.8c0 2.06-1.35 3.73-3.41 3.73-2.07 0-3.63-1.67-3.63-3.73l.03-.36C5.21 7.51 4 10.62 4 14c0 4.42 3.58 8 8 8s8-3.58 8-8C20 8.61 17.41 3.8 13.5.67zM11.71 19c-1.78 0-3.22-1.4-3.22-3.14 0-1.62 1.05-2.76 2.81-3.12 1.77-.36 3.6-1.21 4.62-2.58.39 1.29.59 2.65.59 4.04 0 2.65-2.15 4.8-4.8 4.8z"
                        />
                      </svg>
                      <span
                        class="text"
                      >
                        hot
                      </span>
                    </button>
                  </div>
                  <h4
                    class="title"
                  >
                    Оберіть ваше місто
                  </h4>
                  <button
                    class="root"
                    type="button"
                  >
                    <p
                      aria-hidden="true"
                      class="input"
                    >
                      <span
                        class="text"
                      >
                        kyiv
                      </span>
                      <span
                        class="icon"
                      />
                    </p>
                  </button>
                  <h4
                    class="title"
                  >
                    Оберіть район
                  </h4>
                  <div
                    class="flex"
                  >
                    <button
                      class="chip"
                      title="darnytsia"
                      type="button"
                    >
                      <span
                        class="text"
                      >
                        darnytsia
                      </span>
                    </button>
                    <button
                      class="chip"
                      title="desnianskyi"
                      type="button"
                    >
                      <span
                        class="text"
                      >
                        desnianskyi
                      </span>
                    </button>
                    <button
                      class="chip"
                      title="dniprovskyi"
                      type="button"
                    >
                      <span
                        class="text"
                      >
                        dniprovskyi
                      </span>
                    </button>
                    <button
                      class="chip"
                      title="holosiiv"
                      type="button"
                    >
                      <span
                        class="text"
                      >
                        holosiiv
                      </span>
                    </button>
                    <button
                      class="chip"
                      title="obolonskyi"
                      type="button"
                    >
                      <span
                        class="text"
                      >
                        obolonskyi
                      </span>
                    </button>
                    <button
                      class="chip"
                      title="pecherskyi"
                      type="button"
                    >
                      <span
                        class="text"
                      >
                        pecherskyi
                      </span>
                    </button>
                    <button
                      class="chip"
                      title="podil"
                      type="button"
                    >
                      <span
                        class="text"
                      >
                        podil
                      </span>
                    </button>
                    <button
                      class="chip"
                      title="shevchenkivskyi"
                      type="button"
                    >
                      <span
                        class="text"
                      >
                        shevchenkivskyi
                      </span>
                    </button>
                    <button
                      class="chip"
                      title="solomianskyi"
                      type="button"
                    >
                      <span
                        class="text"
                      >
                        solomianskyi
                      </span>
                    </button>
                    <button
                      class="chip"
                      title="sviatoshynskyi"
                      type="button"
                    >
                      <span
                        class="text"
                      >
                        sviatoshynskyi
                      </span>
                    </button>
                    <button
                      class="chip"
                      title="none"
                      type="button"
                    >
                      <span
                        class="text"
                      >
                        none
                      </span>
                    </button>
                  </div>
                  <h4
                    class="title"
                  >
                    Кількість кімнат
                  </h4>
                  <div
                    class="flex"
                  >
                    <button
                      class="chip"
                      title="one"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M21 10c-1.1 0-2 .9-2 2v3H5v-3c0-1.1-.89-2-2-2s-2 .9-2 2v5c0 1.1.9 2 2 2h18c1.1 0 2-.9 2-2v-5c0-1.1-.9-2-2-2zm-3-5H6c-1.1 0-2 .9-2 2v2.15c1.16.41 2 1.52 2 2.81V14h12v-2.03c0-1.3.84-2.4 2-2.81V7c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                      <span
                        class="text"
                      >
                        one
                      </span>
                    </button>
                    <button
                      class="chip"
                      title="two"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M21 10c-1.1 0-2 .9-2 2v3H5v-3c0-1.1-.89-2-2-2s-2 .9-2 2v5c0 1.1.9 2 2 2h18c1.1 0 2-.9 2-2v-5c0-1.1-.9-2-2-2zm-3-5H6c-1.1 0-2 .9-2 2v2.15c1.16.41 2 1.52 2 2.81V14h12v-2.03c0-1.3.84-2.4 2-2.81V7c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                      <span
                        class="text"
                      >
                        two
                      </span>
                    </button>
                    <button
                      class="chip"
                      title="three"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M21 10c-1.1 0-2 .9-2 2v3H5v-3c0-1.1-.89-2-2-2s-2 .9-2 2v5c0 1.1.9 2 2 2h18c1.1 0 2-.9 2-2v-5c0-1.1-.9-2-2-2zm-3-5H6c-1.1 0-2 .9-2 2v2.15c1.16.41 2 1.52 2 2.81V14h12v-2.03c0-1.3.84-2.4 2-2.81V7c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                      <span
                        class="text"
                      >
                        three
                      </span>
                    </button>
                    <button
                      class="chip"
                      title="four"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M21 10c-1.1 0-2 .9-2 2v3H5v-3c0-1.1-.89-2-2-2s-2 .9-2 2v5c0 1.1.9 2 2 2h18c1.1 0 2-.9 2-2v-5c0-1.1-.9-2-2-2zm-3-5H6c-1.1 0-2 .9-2 2v2.15c1.16.41 2 1.52 2 2.81V14h12v-2.03c0-1.3.84-2.4 2-2.81V7c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                      <span
                        class="text"
                      >
                        four
                      </span>
                    </button>
                    <button
                      class="chip"
                      title="more"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M21 10c-1.1 0-2 .9-2 2v3H5v-3c0-1.1-.89-2-2-2s-2 .9-2 2v5c0 1.1.9 2 2 2h18c1.1 0 2-.9 2-2v-5c0-1.1-.9-2-2-2zm-3-5H6c-1.1 0-2 .9-2 2v2.15c1.16.41 2 1.52 2 2.81V14h12v-2.03c0-1.3.84-2.4 2-2.81V7c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                      <span
                        class="text"
                      >
                        more
                      </span>
                    </button>
                  </div>
                  <h4
                    class="title"
                  >
                    Тип будинку
                  </h4>
                  <div
                    class="flex"
                  >
                    <button
                      class="chip"
                      title="old"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.17 5.7L1 10.48V21h5v-8h4v8h5V10.25z"
                        />
                        <path
                          d="M10 3v1.51l2 1.33L13.73 7H15v.85l2 1.34V11h2v2h-2v2h2v2h-2v4h6V3H10zm9 6h-2V7h2v2z"
                        />
                      </svg>
                      <span
                        class="text"
                      >
                        old
                      </span>
                    </button>
                    <button
                      class="chip"
                      title="new"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M17 11V3H7v4H3v14h8v-4h2v4h8V11h-4zM7 19H5v-2h2v2zm0-4H5v-2h2v2zm0-4H5V9h2v2zm4 4H9v-2h2v2zm0-4H9V9h2v2zm0-4H9V5h2v2zm4 8h-2v-2h2v2zm0-4h-2V9h2v2zm0-4h-2V5h2v2zm4 12h-2v-2h2v2zm0-4h-2v-2h2v2z"
                        />
                      </svg>
                      <span
                        class="text"
                      >
                        new
                      </span>
                    </button>
                  </div>
                  <h4
                    class="title"
                  >
                    Цінова категорія
                  </h4>
                  <div
                    class="flex"
                  >
                    <button
                      class="chip"
                      title="price_one"
                      type="button"
                    >
                      <span
                        class="text"
                      >
                        price_one
                      </span>
                    </button>
                    <button
                      class="chip"
                      title="price_two"
                      type="button"
                    >
                      <span
                        class="text"
                      >
                        price_two
                      </span>
                    </button>
                    <button
                      class="chip"
                      title="price_three"
                      type="button"
                    >
                      <span
                        class="text"
                      >
                        price_three
                      </span>
                    </button>
                    <button
                      class="chip"
                      title="price_four"
                      type="button"
                    >
                      <span
                        class="text"
                      >
                        price_four
                      </span>
                    </button>
                    <button
                      class="chip"
                      title="price_five"
                      type="button"
                    >
                      <span
                        class="text"
                      >
                        price_five
                      </span>
                    </button>
                    <button
                      class="chip"
                      title="price_six"
                      type="button"
                    >
                      <span
                        class="text"
                      >
                        price_six
                      </span>
                    </button>
                    <button
                      class="chip"
                      title="price_seven"
                      type="button"
                    >
                      <span
                        class="text"
                      >
                        price_seven
                      </span>
                    </button>
                  </div>
                  <div
                    class="flex"
                  >
                    <button
                      class="button secondary"
                      type="button"
                    >
                      <span
                        class="children"
                      >
                        Очистити
                      </span>
                    </button>
                    <button
                      class="button primary"
                      type="button"
                    >
                      <span
                        class="children"
                      >
                        Пошук
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </aside>
        </div>
      </div>
    </div>
  </section>
  <footer
    class="footer"
  >
    <div
      class="lg root"
    >
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad eligendi error est fugiat iure maxime molestiae nobis odit, pariatur reiciendis rem sit, unde velit? Earum labore magnam quas quos voluptas!
    </div>
  </footer>
</DocumentFragment>
`;
